core/src/main/java/hudson/cli/DisablePluginCommand.java

```diff
@@ -0,0 +1,135 @@
+
+    @Argument(metaVar = "plugins", required = true, usage = "Plugins to be disabled.")
+    private List<String> pluginNames;
+
+    @Option(name = "-restart", usage = "Restart Jenkins after disabling plugins.")
+    private boolean restart;
+
+    @Override
+    public String getShortDescription() {
+        return Messages.DisablePluginCommand_ShortDescription();
+    }
+
+    @Override
+    protected void printUsageSummary(PrintStream stderr) {
+        super.printUsageSummary(stderr);
+        stderr.println(Messages.DisablePluginCommand_PrintUsageSummary());
+    }
+
+    @Override
+    protected int run() throws Exception {
+        Jenkins jenkins = Jenkins.get();
+        jenkins.checkPermission(Jenkins.ADMINISTER);
+
+        PluginManager manager = jenkins.getPluginManager();
+        int result = 0;
+        boolean somePluginDisabled = false; // to ensure we restart when it needs to
+        for (String pluginName : pluginNames) {
+            switch (disablePlugin(manager, pluginName)) {
+                case DISABLED:
+                    if(!somePluginDisabled) somePluginDisabled = true;
+                    break;
+                case NOT_DISABLED_DEPENDANTS:
+                    if(result != 16) result = 16; // Custom error code to indicate one plugin, at least, can't be disabled
+            }
+        }
+
+        if (restart && somePluginDisabled) {
+            jenkins.safeRestart();
+        }
+
+        return result;
+    }
+
+    /**
+     * Try to disable a  plugin.
+     * @param manager The PluginManager.
+     * @param shortName The name of the plugin to disable.
+     * @return The result of the disabling of this plugin. See {@link DisablingStatus}
+     * @throws IOException An exception disabling the plugin. See {@link PluginWrapper#disable()}
+     */
+    private DisablingStatus disablePlugin(PluginManager manager, String shortName) throws IOException {
+        PluginWrapper plugin = manager.getPlugin(shortName);
+        if (plugin == null) {
+            throw new IllegalArgumentException(Messages.DisablePluginCommand_NoSuchPlugin(shortName)); // exit with 3
+        }
+
+        if (!plugin.isEnabled()) {
+            stdout.format(Messages.DisablePluginCommand_Already_Disabled(shortName));
+            stdout.println();
+            return DisablingStatus.ALREADY_DISABLED;
+        }
+
+        Set<String> dependants = plugin.getDependants();
+        for (String dependant : dependants) {
+            PluginWrapper dependantPlugin = manager.getPlugin(dependant);
+            if (dependantPlugin != null && dependantPlugin.isEnabled()) {
+                stderr.format(Messages.DisablePluginCommand_Plugin_Has_Dependant(shortName, dependant));
+                stderr.println();
+                return DisablingStatus.NOT_DISABLED_DEPENDANTS;
+            }
+        }
+
+        plugin.disable();
+        stdout.format(Messages.DisablePluginCommand_Plugin_Disabled(plugin.getShortName()));
```

Comment:
I'm not sure about printing the messages here. I think it makes more sense having the disable logic here and then printing the messages anywhere else.

Reply:
Could you explain it please?

Reply:
To me it makes more sense than this command would return the result of disabling the plugin, and the place where it is called react to that result and prints the corresponding message.

Reply:
Maybe you're proposing like a final summary? It could be good idea.